File: Bit Manipulation/338.Counting Bits.cpp
class Solution {
    public:
    string binaryrep(int m){
        string s="";
        while(m>0){
            s+=((m%2)+'0');
            m/=2;
        }
        reverse(s.begin(),s.end());
        return s;
    }
    int count(string ss){
        int ans=0;
        for(int i=0;i<ss.length();i++){
            if(ss[i]=='1'){
                ans++;
            }
        }
        return ans;
    }
public:
    vector<int> countBits(int n) {
        vector<int> ans;
        for(int i=0;i<=n;i++){
            string ss=binaryrep(i);
            int cnt=count(ss);
            ans.push_back(cnt);
        }
        return ans;
    }
};

File: Bit Manipulation/338.Counting Bits.java
class Solution {
    public String binaryrep(int m){
        if(m==0){
            return "0";
        }
        String s="";
        while(m>0){
            s+=(char)((m%2)+'0');
            m/=2;
        }
        String rev=new StringBuilder(s).reverse().toString();
        return rev;
    }
    public int count(String ss){
        int ans=0;
        for(int i=0;i<ss.length();i++){
            if(ss.charAt(i)=='1'){
                ans++;
            }
        }
        return ans;
    }
    public int[] countBits(int n) {
        int ans[]=new int[n+1];
        for(int i=0;i<=n;i++){
            String s=binaryrep(i);
            int cnt=count(s);
            ans[i]=cnt;
        }
        return ans;
    }
}
